MekHQ/src/mekhq/campaign/mission/AtBDynamicScenarioFactory.java

```diff
@@ -1099,6 +1108,64 @@ private static Entity generateTransportedBAUnit(UnitGeneratorParameters params,
 
         return transportedUnits;
     }
+    
+    /**
+     * Worker function that generates a battle armor unit to attach to a unit of clan mechs
+     */
+    public static List<Entity> generateBAForNova(AtBScenario scenario, List<Entity> starUnits, String factionCode, int skill, int quality, Campaign campaign) {
+        List<Entity> transportedUnits = new ArrayList<>();
+        
+        // determine if this should be a nova
+        // if yes, then pick the fastest mech and load it up, adding the generated BA to the transport relationships.
+        
+        // non-clan forces and units that aren't stars don't become novas
+        if(!Faction.getFaction(factionCode).isClan() && (starUnits.size() != 5)) {
+            return transportedUnits;
+        }
+        
+        // logic copied from AtBScenario.addStar() to randomly determine if the given unit is actually going to be a nova
+        // adjusted from 11/8 to 8/6 (distribution of novas in newest AtB doc is a lot higher) so that players actually encounter novas
+        // whatever CBS is still gets no novas, so there
+        int roll = Compute.d6(2);
+        int novaTarget = 8;
+        if (factionCode.equals("CHH") || factionCode.equals("CSL")) {
+            novaTarget = 6;
+        } else if (factionCode.equals("CBS")) {
+            novaTarget = 13;
+        }
+        
+        if(roll < novaTarget) {
+            return transportedUnits;
+        }
+        
+        Entity actualTransport = null;
+        for(Entity transport : starUnits) {
```

Comment:
We're picking the slowest one assuming its the heaviest?

Reply:
Whoops, I wanted to put them on the fastest one.

Reply:
Looking at it again, actual transport is the "current fastest". If actualtransport.getMP is less than the currently evaluated transport.getMP, then the currently evaluated transport becomes the actual transport by virtue of being faster.